experimental/Schemes/duValSing.jl

```diff
@@ -0,0 +1,262 @@
+- `T[2]` represents a point at which some non-du-Val singularity is present
+- `T[3]` is the empty tuple
+- `T[4] = 1`
+
+**Note**: For the ideal I in a ring R, dim(R/I) = 0 is asserted
+
+# Example:
+```jldoctest
+julia> R,(x,y,z,w) = QQ["x","y","z","w"]
+(Multivariate polynomial ring in 4 variables over QQ, QQMPolyRingElem[x, y, z, w])
+
+julia> I = ideal(R,[w,x^2+y^3+z^4])
+ideal(w, x^2 + y^3 + z^4)
+
+julia> Rq,_ = quo(R,I)
+(Quotient of multivariate polynomial ring by ideal with 2 generators, Map from
+Multivariate polynomial ring in 4 variables over QQ to Rq defined by a julia-function with inverse)
+
+julia> J = ideal(R,[x,y,z,w])
+ideal(x, y, z, w)
+
+julia> decide_du_val_singularity(X,J)
+1-element Vector{Tuple{Bool, MPolyIdeal{QQMPolyRingElem}, Tuple{Symbol, Int64}, Int64}}:
+ (1, ideal(x, y, z, w), (:E, 6), 1)
+
+""" 
+function decide_du_val_singularity(X::AbsSpec{<:Field,<:Any},I::MPolyIdeal)
+  OOX = OO(X)
+  dim(X) == 2 || error("Scheme not of dimension 2")
+  J = modulus(OOX)
+  !isone(J) || error("Scheme is empty")
+  !iszero(J) || return true                  ## X smooth
+
+  R = base_ring(I)
+  kk = base_ring(R)
+  characteristic(kk) == 0 || error("only available in characteristic zero")
+  base_ring(OOX) === R || error("base rings need to be identical")
+
+  dim(I) == 0 || error("second argument is not a point")
+  if get_attribute(I,:is_absolutely_prime, false)
+    li = _check_duval_at_point(J,I)
+    return [(li[1], I, li[2],1)]
+  end
+
+  result_vector = []
+  decomp = absolute_primary_decomposition(I)
+  for (_,_,I2,mult) in decomp
+    set_attribute!(I2,:is_absolutely_prime,true)
+    ## pass to algebraic extension
+    r_changed = base_ring(I2)
+    kk = coefficient_ring(r_changed)
+    J_changed = ideal(r_changed,  [change_coefficient_ring(kk,a, parent = r_changed) for a=gens(J)])
+    tempvec = decide_du_val_singularity(Spec(quo(r_changed,J_changed)[1]),I2)
+    for x in tempvec
+      x[1] || return [x]
+      x = (x[1],x[2],x[3],mult)
+      push!(result_vector,x)
+    end
+  end
+
+  return result_vector
+end
+
+@doc raw"""
+    _check_du_val_at_point(IX:Ideal,Ipt::Ideal)
+
+Returns a tuple T with the following data:
+- T[1]::Bool has V(IX) at most a du Val singularity at V(Ipt)
+- T[2]::Tuple Type of du Val singularity at V(Ipt)
+
+**Note**: Assumes Ipt to be absolutely irreducible.
+
+**Note**: Do not call directly, only via the higher level functions is_du_Val_singularity and decide_du_Val_singularity.
+
+"""
+function _check_duval_at_point(IX::Ideal,Ipt::Ideal)
+  R = base_ring(IX)
+  R == base_ring(Ipt) || error("basering mismatch")
+  kk = base_ring(R)
+  characteristic(kk) == 0 || error("only available in characteristic zero")
+ 
+  JM = jacobi_matrix(gens(IX))
+
+  smooth = (:A,0)
+
+  ## localize at point
+  a = rational_point_coordinates(Ipt)
+  U = complement_of_point_ideal(R,a)
+  RL, loc_map = localization(R,U)
+  IX_loc = loc_map(IX)
+  JM_loc =  map(x ->loc_map(x), JM[:,:])
+
+  if !all(iszero(a))
```

Comment:
I would suggest either
```suggestion
  if !all(iszero,a)
```
or
```suggestion
  if !iszero(a)
```

Reply:
Thanks for the hint.